### PR TITLE
Feat: Notes api calls(save,get,delete,update)

### PR DESCRIPTION
**Pull request by Rohitprasad83**
## Ignore:-
- the console.logs, will replace them with Toast messages after this PR.
- will add more attributes in the note object later like tags.

##GIF

chrome_pJjYGrDQLK

### CODE REVIEW

File: src/components/main/MainContent.jsx [hunk 1]
```diff
@@ -1,68 +1,134 @@
...
+import { colors } from './colors'
+
 function MainContent() {
+  const { note, noteDispatch, notes, setNotes } = useNote()
+  const [showPalette, setShowPalette] = useState(false)
+  const [createNote, setCreateNote] = useState(false)
+
+  const { encodedToken } = useAuth()
+
+  const saveNoteHandler = async e => {
```
Comment (ishitakeshawani) [inline src/components/main/MainContent.jsx:14]: You can add this function to some other file to keep it clean.

File: src/components/main/Note.jsx [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+  const { encodedToken } = useAuth()
+  const [openEdit, setOpenEdit] = useState(false)
+  const [showPalette, setShowPalette] = useState(false)
+  const [editNote, editNoteDispatch] = useReducer(createNoteReducer, {
+    _id: _id,
+    title: title,
+    body: body,
+    color: color,
+  })
+  const deleteNoteHandler = async e => {
```
Comment (ishitakeshawani) [inline src/components/main/Note.jsx:21]: same for it.

File: src/components/main/Note.jsx [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+            className="input__body"
+            value={editNote.body}
+            onChange={e =>
+              editNoteDispatch({ type: 'BODY', payload: e.target.value })
+            }></textarea>
+          <div className="text__lg note__bottom">
+            <i
+              className="fa-solid fa-palette input__icons palette__container"
+              onClick={() => setShowPalette(!showPalette)}></i>
+            {showPalette && (
```
Comment (ishitakeshawani) [inline src/components/main/Note.jsx:82]: You can make component for this palette.

**Review by ishitakeshawani — APPROVED** [review]
I liked the UI of this.
Good use of different contexts.
Added suggetsions.

File: src/components/main/MainContent.jsx [hunk 4]
```diff
@@ -1,68 +1,134 @@
...
+    try {
+      const response = await axios.post(
+        '/api/notes',
+        {
+          note,
+        },
+        {
+          headers: { authorization: encodedToken },
+        }
+      )
```
Comment (hrishikeshmane) [inline src/components/main/MainContent.jsx:25]: format code with prettier.

File: src/components/main/MainContent.jsx [hunk 5]
```diff
@@ -1,68 +1,134 @@
...
+      console.log(response)
```
Comment (hrishikeshmane) [inline src/components/main/MainContent.jsx:26]: remove console logs before pushing.

**Review by hrishikeshmane — APPROVED** [review]

File: src/components/main/MainContent.jsx (hunk 4 above)
Comment (Rohitprasad83) [reply src/components/main/MainContent.jsx:25]: it is formatted using prettier, might be some prettier settings.

File: src/components/main/MainContent.jsx (hunk 5 above)
Comment (Rohitprasad83) [reply src/components/main/MainContent.jsx:26]: added a heading in PR request to ignore console.logs, will replace them with Toast.

File: src/components/main/MainContent.jsx (hunk 1 above)
Comment (Rohitprasad83) [reply src/components/main/MainContent.jsx:14]: sure will do it

File: src/components/main/Note.jsx (hunk 3 above)
Comment (Rohitprasad83) [reply src/components/main/Note.jsx:82]: okay will do it

**Pull request merged**